src/VisualStudio/Core/Def/Implementation/Options/RoamingVisualStudioProfileOptionPersister.cs

```diff
@@ -132,9 +132,23 @@ public bool TryFetch(OptionKey optionKey, out object value)
                 }
                 else
                 {
-                    value = optionKey.Option.DefaultValue;
+                    value = null;
+                    return false;
                 }
             }
+            else if (optionKey.Option.Type == typeof(bool) && value is int)
+            {
+                // TypeScript used to store some booleans as integers. We now handle them properly for legacy sync
+                // scenarios.
+                value = ((int)value != 0);
+                return true;
+            }
+            else if (value != null && optionKey.Option.Type != value.GetType())
```

Comment:
Are enums properly handled somewhere?  I seem to recall that that messes with things as the Enum.Type won't match the Int.type that things are stored as.